Updating variable names that use "json" nomenclature

jasonkylefrank opened the pull request:
<!-- 👀 Thanks for opening a PR! Read comments like this one to get your PR merged faster. -->
#### Summary
Updating variable names that use "json" where the data is not in json form so that new learners (and old learners like me) are not misled. 😀

#### Motivation
I'm teaching my 2 sons how to program, and one of the lessons I'm trying to teach them is to always ask yourself, "What kind of thing is this?" (which in-turn tells you what you can do with it, etc.).  Taking that lesson, they look at these examples and see that a piece of code like:
```
const json = await response.json();
```
...and assume that the `json` variable holds an object in json form.


#### Supporting details
NOTE: This commit is probably a conversation-starter.  If you agree with the spirit of the changes, then other variable names in this file might also need to be update for consistency.  

ISSUE BEING ADDRESSED:  
`response.json()` returns a regular JS object (not json).  See: https://developer.mozilla.org/en-US/docs/Web/API/Response/json.  So for new learners (and old learners like me!), seeing a variable named "json" is quite misleading if it is not actually json.  For now, I just changed a few lines in the Async/Await section that used the variable name "json" (but we may also want to update other sections of this file for consistency reasons).


#### Metadata
<!-- ✅ Check a box if applicable, like this: [x]

This PR…
-->
- [ ] Adds a new document
- [ ] Rewrites (or significantly expands) a document
- [x] Fixes a typo, bug, or other error

<!-- 👷‍♀️ After submitting, review the results of the "Checks" tab! -->

jasonkylefrank committed:
Updating variable names that use "json" nomenclature

NOTE: This commit is probably a conversation-starter.  If you agree with the spirit of the changes, then other variable names in this file might also need to be update for consistency.  

ISSUES BEING ADDRESSED:  
(1) response.json() returns a regular JS object (not json).  See: https://developer.mozilla.org/en-US/docs/Web/API/Response/json.  So for new learners (and old learners like me!), seeing a variable named "json" is quite misleading if it is not actually json.  For now, I just changed a few lines in the Async/Await section that used the variable name "json" (but we may also want to update other sections of this file for consistency reasons).

(2) The last example under Async/Await used a ".then()" in an example which is focused on using async/await.  For clarity to new learners, I changed it to use await.

jasonkylefrank committed:
Fixed last await part to use .then()

Can't use await outside the context of an async function, so I changed that example back to .then().

wbamberg commented on files/en-us/learn/javascript/asynchronous/promises/index.md:
```diff
@@ -333,16 +333,17 @@ async function fetchProducts() {
-jsonPromise.then((json) => console.log(json[0].name));
+// Note: We can't use `await` outside the context of an async function. So we'll use `.then()` here for simplicity.
+const promise = fetchProducts();
+promise.then(data => console.log(data[0].name);
```
```suggestion
promise.then(data => console.log(data[0].name));
```

jasonkylefrank replied:
@Phosra Thanks. Regarding the comment that I added on line 344 and your note about [Top Level Await, ](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/await#top_level_await), I'm not sure the best way to include that extra aspect without cluttering up the comment.  I also thought about adding a sentence or two about it in a paragraph below the code, but I wasn't sure if that would seem like too much of a tangent or how much to say about it on this page...

Part of the issue regarding adding that aspect into the explanation as to why we're using `.then()` in an example that is supposed to be about `await`, is that, the way I understand it, we can only use top-level await if the code is a module.  In this particular example, I don't believe we have a module (yet)... 

So should we turn it into a module so that the entire example uses `await`?
Should we add a short paragraph under the code example to further explain the issue?
Or something else?

jasonkylefrank replied:
@Phosra Ok, should we resolve this particular conversation with the code as-is then?
(I also just committed a change to address the "let's fix the rest of the file for consistency" aspect, so we can converse separately about that.)

wbamberg replied:
As I understood it, top level await is only available in modules*. Since this is introductory material we should not expect people to learn about modules at this point.

But the comment "We can't use `await` outside the context of an async function" is problematic, because it's wrong. Having said that we also say "this magic only works inside the async function" just above, which is also wrong.

So I think we do need to expand on this a little here. But let's do it in a separate PR, it's unrelated to this.

* although it seems like it works in the console as well.

wbamberg reviewed (changes requested):
Yes, I'm happy with this suggestion. Thanks!

In one place a bracket went missing, and I think we should update the whole page in this PR before merging.

jasonkylefrank committed:
Fixed missing closing paren

Co-authored-by: wbamberg <will@bootbonnet.ca>

jasonkylefrank committed:
Replaced other `json` names where appropriate

Using the same idea from my original commit, I changed other examples in this file where a variable named `json` is not appropriate.

jasonkylefrank commented:
@wbamberg I made a pass at updating the rest of the page for consistency regarding the `json` variable naming change.  Anything else you'd like to see done?

wbamberg reviewed (approved):
👍 thank you @jasonkylefrank !

jasonkylefrank commented:
@wbamberg You're welcome, glad to see it improve!
Also, forgive my ignorance, but now that it's been merged into `main`, when does it get deployed?

wbamberg commented:
>  now that it's been merged into main, when does it get deployed?

A day or so, I think :).
